### ISSUE TITLE
Cannot use the Server Version console option to upgrade from 7.89 to the stable version 9.52

### ISSUE THREAD

**Issue by cahsdhelpdesk** [Ylianst/MeshCentral#3590]
Good afternoon.
We are currently on 7.89 and I waited until there was a "stable" release to upgrade to.
Now that I see that 9.52 is stable, I cannot upgrade to that version using the console.
Is this by design and if so, is there another way to upgrade without losing all the devices and config I have now?

Regards

Jeff

**Comment by Ylianst** [comment]
Not sure what you mean by "console".

You can update from the Windows or Linux shell if it's not working from the web site. First, make sure you backup your database and "meshcentral-data" folder offsite. Then, stop the server, rename "node_modules" to "node_modules_bak" and get the latest version using:

```
npm install meshcentral@0.9.52
node node_modules/meshcentral
```

This will run the stable version of MeshCentral and install any missing dependencies. Once it's running ok, hit ctrl-c and run the server again in the background.

If you want to go back, just put the old `node_modules` back and run the server again.

Hope that helps,
Ylian

**Comment by cahsdhelpdesk** [comment]
Good morning and thanks for your reply.
When I run the npm command I receive the following.

WARN engine meshcentral@0.9.52: wanted: {"node":">=10.0.0"} (current: {"node":"8.10.0","npm":"3.5.2"})
npm ERR! Linux 4.15.0-163-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install" "meshcentral@0.9.52"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! code EMISSINGARG

npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at andLogAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:31:3)
npm ERR! typeerror     at fetchPackageMetadata (/usr/share/npm/lib/fetch-package-metadata.js:51:22)
npm ERR! typeerror     at resolveWithNewModule (/usr/share/npm/lib/install/deps.js:456:12)
npm ERR! typeerror     at /usr/share/npm/lib/install/deps.js:457:7
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror     at /usr/share/npm/lib/fetch-package-metadata.js:37:12
npm ERR! typeerror     at addRequestedAndFinish (/usr/share/npm/lib/fetch-package-metadata.js:82:5)
npm ERR! typeerror     at returnAndAddMetadata (/usr/share/npm/lib/fetch-package-metadata.js:117:7)
npm ERR! typeerror     at pickVersionFromRegistryDocument (/usr/share/npm/lib/fetch-package-metadata.js:134:20)
npm ERR! typeerror     at /usr/share/npm/node_modules/iferr/index.js:13:50
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <http://github.com/npm/npm/issues>
WARN engine mkdirp@1.0.4: wanted: {"node":">=10"} (current: {"node":"8.10.0","npm":"3.5.2"})

Any thoughts?

**Comment by rapi3** [comment]
see warn line; your node & npm version are to old and unsupported by current version of MescCentral.

My sugestion is to install latest node LTS and npm
I just updated on linux to:
node v16.13.2
npm 8.4.0

**Comment by cahsdhelpdesk** [comment]
Thank you, that makes sense however when I try to upgrade this is returned.

Reading package lists... Done
Building dependency tree
Reading state information... Done
npm is already the newest version (3.5.2-0ubuntu4).

It is as if there is a "restriction" on what the latest version is, downloaded, or can be installed?

Any thoughts?

**Comment by rapi3** [comment]
for latest ubuntu server 20.4 today I had to do manual install of latest node LTS, version from repo it was old and I had ptoblems with that:

( unistall/install nodejs and npm only you need to do it as root )
- uninstall node and purge... 
- download latest LTS for linux: wget https://nodejs.org/dist/v16.13.2/node-v16.13.2-linux-x64.tar.xz
- create dir for node: mkdir /usr/local/lib/nodejs/bin/nodejs
- extract tar xf node-v16.13.2-linux-x64.tar.xz -C /usr/local/lib/nodejs
- create link for node: ln -s /usr/local/lib/nodejs/bin/node /usr/bin/node
- check node -v and npm -v

https://github.com/nodejs/help/wiki/Installation

you can also try with nvm:
https://github.com/nvm-sh/nvm

**Comment by cahsdhelpdesk** [comment]
Thank you. But I think I prefer to find out why the package will not update before I uninstall anything.
Can anyone tell me why this error would occur?
npm is already the newest version (3.5.2-0ubuntu4).

**Comment by rapi3** [comment]
The message translate that in repo you use that is the latest version you have installed.
What version of linux you use ?
If your linux it is a EOL version then old repo for that version are not update any more.

**Comment by cahsdhelpdesk** [comment]
I should have included this in the beginning, Ubuntu 18.04 LTS Server.

**Comment by rapi3** [comment]
Just to be sure save a backup of your MeshCentral server on another computer.
Your only chance is to install manual a newer version by hand as explained https://github.com/nodejs/help/wiki/Installation
And prepare yourself in the future for a server upgrade.
Good luck.

**Comment by cahsdhelpdesk** [comment]
Good news.
The warning above, WARN engine meshcentral@0.9.52: wanted: {"node":">=10.0.0"} (current: {"node":"8.10.0","npm":"3.5.2"}) which you pointed out needed to be upgraded. 
All my attempts to upgrade them failed.

However, on an educated hunch (after backing up everything) I performed an in-place upgrade to Ubuntu 20.04.2 LTS.
https://www.cyberciti.biz/faq/upgrade-ubuntu-18-04-to-20-04-lts-using-command-line/
This upgraded nodejs to 10.19.0 and npm to 6.14.4.
After the reboot, (crossing my fingers) I logged into MeshCentral, I able to upgrade to 9.52 using the web console.

This may not be the path for others but if so, perhaps this will help someone else in the future.
